mesonbuild/minstall.py

```diff
@@ -452,6 +458,29 @@ def do_copyfile(self, from_file: str, to_file: str,
         append_to_log(self.lf, to_file)
         return True
 
+    def do_symlink(self, target: str, link: str, full_dst_dir: str) -> bool:
+        abs_target = target
+        if not os.path.isabs(target):
+            abs_target = os.path.join(full_dst_dir, target)
+        if not os.path.exists(abs_target):
+            raise RuntimeError(f'Tried to install symlink to missing file {abs_target}')
+        if os.path.exists(link):
+            if not os.path.islink(link):
+                raise RuntimeError(f'Destination {link!r} already exists and is not a symlink')
+            self.remove(link)
+        if not self.printed_symlink_error:
+            self.log(f'Installing symlink {link} to {target}')
```

Comment:
I upgraded a local project to use the new function, and gave it a test run. Noticed one teensy tiny little UI issue:

```
[...]
Installing doc/alpm-hooks.5 to /tmp/builddir/pkg/usr/share/man/man5
[...]
Installing doc/repo-add.8 to /tmp/builddir/pkg/usr/share/man/man8
[...]
Installing symlink /tmp/builddir/pkg/usr/share/man/man8/repo-remove.8 to repo-add.8
Installing symlink /tmp/builddir/pkg/usr/share/man/man5/pacman-hooks.5 to alpm-hooks.5
[...]
```

The link and target are swapped, IMO. The last position in each log line, "to XXX", is traditionally the on-disk node that is being created.

Reply:
I think here again the issue raised by @jpakkane hits again, I really do not get from my own message what is the link and what is the target.

Maybe `Installing symlink {os.path.basename(link)} pointing to {target} to {full_dst_dir}`?

Reply:
Maybe `Installing symlink with target {target} to {link}`?

Reply:
I still think it makes sense to keep the `pointing to` syntax because it is more obvious and used in the function. Should `Installing symlink pointing to {target} to {link}` work?

Reply:
Changed in last commit. Let me know if it would need further changes.